buzzwindow.cpp

```diff
@@ -67,18 +69,26 @@ void BuzzWindow::MessageReceived(BMessage* message)
    		{// Look for a ref in the message
 		   	entry_ref ref;
 			BPath path;
-   			if(message->FindRef("refs", &ref) == B_OK)
-   			{
+			if(message->FindRef("refs", &ref) == B_OK) {
 				BEntry Entry(&ref);
 				Entry.GetPath(&path);
 				if((Entry.GetRef(&ref) == B_OK))
 				{
-					MyClock->SetSound(new BFileGameSound(&ref, false));
-					SoundName->SetText(ref.name);
+					BFileGameSound soundFile(&ref, false);
+					if (soundFile.InitCheck() == B_OK) {
+						MyClock->SetSound(&soundFile);
+						SoundName->SetText(ref.name);
+					} else {
+						BAlert* not_supported = new BAlert(
+							"Error",
+							"This is not a supported sound file.",
+							"OK", NULL, NULL, B_WIDTH_AS_USUAL,
+							B_OFFSET_SPACING, B_WARNING_ALERT);
+						not_supported->Go();
+					}
 				}
```

Comment:
Delete it.

Reply:
I prefer [this](https://github.com/HaikuArchives/WakeUp/pull/14#discussion_r366681410) to the above though.